service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java

```diff
@@ -350,6 +360,78 @@ public void explainPolicies() throws Exception {
     assertEquals(0, result.getExplanation().size());
   }
 
+  @Test
+  @EnabledIf(expression = "${feature.tps-enabled}", loadContext = true)
+  public void mergeCheck_sameWorkspace() throws Exception {
+    ApiWsmPolicyUpdateResult result =
+        mergeCheck(userAccessUtils.defaultUserAuthRequest(), workspace.getId(), workspace.getId());
+
+    assertEquals(0, result.getConflicts().size());
+  }
+
+  @Test
+  @EnabledIf(expression = "${feature.tps-enabled}", loadContext = true)
+  public void mergeCheck_differentRegion() throws Exception {
+    // Create workspace with US region constraint.
+    ApiCreateWorkspaceRequestBody targetRequest =
```

Comment:
ah you can extract l376 - l396 to a helper method createWorkspaceWithRegionConstraint(String regionName)